### PR TITLE
Verify If It's Possible to Deploy Pachyderm Server and Created Teardown

### PR DESCRIPTION
**Pull request by risusing**

### CODE REVIEW

File: tests/Tests/600__ai_apps/670__pachyderm/670__pachyderm.robot [hunk 1]
```diff
@@ -2,23 +2,28 @@
...
 
 
+
 *** Test Cases ***
 Verify Pachyderm Can Be Installed Using OpenShift Console
-    [Tags]    Tier2    ODS-1137
+    [Tags]      Tier2
+    ...         ODS-1137    ODS-1138
     Check And Install Operator in Openshift    ${pachyderm_container_name}    ${pachyderm_appname}
+    Create Tabname Instance For Installed Operator        ${pachyderm_container_name}   ${pachyderm_container_name}     redhat-ods-applications
```
Comment (bdattoma) [inline tests/Tests/600__ai_apps/670__pachyderm/670__pachyderm.robot:23]: I think there is a missing step which is reported in ODS-1138, that is the secret creation

File: tests/Resources/Page/ODH/AiApps/Pachyderm.robot [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+*** Keywords ***
+Uninstall Pachyderm Operator
+    [Documentation]    Uninstall pachyderm operator and it's realted component
+    Go To    ${OCP_CONSOLE_URL}
+    Maybe Skip Tour
+    Delete Tabname Instance For Installed Operator    ${pachyderm_operator_name}    Pachyderm
+    ...    redhat-ods-applications
+    Uninstall Operator    ${pachyderm_operator_name}
+    OpenShiftCLI.Delete    kind=ImageStream    namespace=redhat-ods-applications
```
Comment (bdattoma) [inline tests/Resources/Page/ODH/AiApps/Pachyderm.robot:15]: is it necessary?

**Comment by risusing** [comment]
For Teardown I have created this keyword. And in similar fashion of Aikit test file. And for Teardown it's necessary.

File: tests/Resources/Page/OCPDashboard/Page.robot [hunk 3]
```diff
@@ -45,3 +53,18 @@ Search Last Item Instance By Title in OpenShift Table
   Click Button    xpath://*/th[@data-label='Created']/button  # asc order
   Click Button    xpath://*/th[@data-label='Created']/button  # desc order
 
+Delete Project By Name
+  [Arguments]  ${project_name}
+  Menu.Navigate To Page   Home    Projects
+  Sleep     2
+  Input Text    //input[@data-test-id="item-filter"]    ${project_name}
+  Sleep     5
```
Comment (bdattoma) [inline tests/Resources/Page/OCPDashboard/Page.robot:61]: could you find an alternative to all of these sleeps? maybe use of wait until

File: tests/Tests/600__ai_apps/670__pachyderm/670__pachyderm.robot [hunk 4]
```diff
@@ -2,23 +2,30 @@
...
 *** Test Cases ***
 Verify Pachyderm Can Be Installed Using OpenShift Console
-    [Tags]    Tier2    ODS-1137
+    [Tags]      Tier2
+    ...         ODS-1137    ODS-1138
     Check And Install Operator in Openshift    ${pachyderm_container_name}    ${pachyderm_appname}
+    Create Project      pachyderm
+    Create Pachyderm AWS-Secret
+    Create Tabname Instance For Installed Operator        ${pachyderm_container_name}   ${pachyderm_container_name}     pachyderm
+    Wait Until Keyword Succeeds     120     1       Wait Until Status Is Running
```
Comment (bdattoma) [inline tests/Tests/600__ai_apps/670__pachyderm/670__pachyderm.robot:25]: could you move the wait logic inside `Wait Until Status Is Running`?

File: tests/Resources/Page/OCPDashboard/Page.robot [hunk 5]
```diff
@@ -26,6 +26,14 @@ Maybe Click Show Default Project Button
      END
   END
 
+Create Project
```
Comment (jgarciao) [inline tests/Resources/Page/OCPDashboard/Page.robot:29]: Missing [Documentation]

File: tests/Resources/Page/OCPDashboard/Page.robot [hunk 6]
```diff
@@ -45,3 +53,16 @@ Search Last Item Instance By Title in OpenShift Table
   Click Button    xpath://*/th[@data-label='Created']/button  # asc order
   Click Button    xpath://*/th[@data-label='Created']/button  # desc order
 
+Delete Project By Name
```
Comment (jgarciao) [inline tests/Resources/Page/OCPDashboard/Page.robot:56]: Missing [Documentation]

File: tests/Tests/600__ai_apps/670__pachyderm/670__pachyderm.robot [hunk 7]
```diff
@@ -2,23 +2,30 @@
...
 
 
 *** Variables ***
 ${pachyderm_container_name}     Pachyderm
+${pachyderm_operator_name}      Pachyderm
 ${pachyderm_appname}            pachyderm
 
 
 *** Test Cases ***
 Verify Pachyderm Can Be Installed Using OpenShift Console
```
Comment (jgarciao) [inline tests/Tests/600__ai_apps/670__pachyderm/670__pachyderm.robot:18]: Missing [Documentation]

File: tests/Tests/600__ai_apps/670__pachyderm/670__pachyderm.robot [hunk 8]
```diff
@@ -2,23 +2,30 @@
...
-    [Tags]    Tier2    ODS-1137
+    [Tags]      Tier2
+    ...         ODS-1137    ODS-1138
     Check And Install Operator in Openshift    ${pachyderm_container_name}    ${pachyderm_appname}
+    Create Project      pachyderm
```
Comment (jgarciao) [inline tests/Tests/600__ai_apps/670__pachyderm/670__pachyderm.robot:22]: I think lines 22-25 should be in a keyword called `Deploy Pachyderm Server`  (or similar). If you want to split ODS-1137 and ODS-1138 in two different tests you can follow the pattern in https://github.com/red-hat-data-services/ods-ci/blob/522735d488cd8c54c71e38b860f8917998ea2944/tests/Tests/600__ai_apps/640__rhoam/640__rhoam_installation.robot#L20

File: tests/Tests/600__ai_apps/670__pachyderm/670__pachyderm.robot [hunk 9]
```diff
@@ -29,3 +36,9 @@ Dashboard Test Setup
 
 Dashboard Test Teardown
     Close All Browsers
+
+Wait Until Status Is Running
+    Wait Until Keyword Succeeds     120     1       Element Text Should Be    //span[@data-test="status-text"]      Running
+
+Create Pachyderm AWS-Secret
```
Comment (jgarciao) [inline tests/Tests/600__ai_apps/670__pachyderm/670__pachyderm.robot:43]: Missing [Documentation]

File: tests/Resources/Page/OCPDashboard/Page.robot [hunk 10]
```diff
@@ -26,6 +26,15 @@ Maybe Click Show Default Project Button
      END
   END
 
+Create Project
+    [Documentation]     This keyword is use to create a project in OCP Console.
```
Comment (bdattoma) [inline tests/Resources/Page/OCPDashboard/Page.robot:30]: could you remove "This Keyword etc" and write like "Create a project in OCP". Same for other docs

File: tests/Resources/Page/OCPDashboard/Page.robot [hunk 11]
```diff
@@ -26,6 +26,15 @@ Maybe Click Show Default Project Button
...
+    [Arguments]  ${project_name}
+    Menu.Navigate To Page   Home    Projects
+    Click Button    xpath://button[@id="yaml-create"]
+    Input Text      xpath://input[@id="input-name"]     ${project_name}
+    Click Button    xpath://button[@id="confirm-action"]
+    Sleep   2
```
Comment (bdattoma) [inline tests/Resources/Page/OCPDashboard/Page.robot:36]: why sleeping?

File: tests/Resources/Page/ODH/AiApps/Pachyderm.robot [hunk 12]
```diff
@@ -0,0 +1,18 @@
+*** Settings ***
+Documentation       Resource file for pachyderm operator
+Library             OpenShiftCLI
+Library             SeleniumLibrary
+
+
+*** Keywords ***
+Uninstall Pachyderm Operator
+    [Documentation]    Uninstall pachyderm operator and it's realted component
```
Comment (bdattoma) [inline tests/Resources/Page/ODH/AiApps/Pachyderm.robot:9]: typo: it's instead of its

**Comment by risusing** [comment]
Jenkins Job : https://opendatascience-jenkins-csb-rhods.apps.ocp-c1.prod.psi.redhat.com/job/rhods-ci-pr-test/609/robot/report/files/log.html
Result : Passed

**Comment by bdattoma** [comment]
lgtm, other than some small comments

File: tests/Resources/Page/ODH/AiApps/Pachyderm.robot [hunk 13]
```diff
@@ -0,0 +1,18 @@
+*** Settings ***
+Documentation       Resource file for pachyderm operator
+Library             OpenShiftCLI
```
Comment (jgarciao) [inline tests/Resources/Page/ODH/AiApps/Pachyderm.robot:3]: Remove this

File: tests/Resources/Page/OCPDashboard/Page.robot (hunk 3 above)
Comment (bdattoma) [reply tests/Resources/Page/OCPDashboard/Page.robot:62]: ?

**Review by bdattoma — APPROVED** [review]
lgtm

**Review by jgarciao — APPROVED** [review]
lgtm

**Pull request merged**